examples/controlnet/README_flux.md

```diff
@@ -0,0 +1,177 @@
+{"image_path": "xxx", "caption": "xxx", "control_path": "xxx"}
+{"image_path": "xxx", "caption": "xxx", "control_path": "xxx"}
+```
+
+
+## Training
+
+Our training examples use two test conditioning images. They can be downloaded by running
+
+```sh
+wget https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/diffusers/controlnet_training/conditioning_image_1.png
+wget https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/diffusers/controlnet_training/conditioning_image_2.png
+```
+
+Then run `huggingface-cli login` to log into your Hugging Face account. This is needed to be able to push the trained ControlNet parameters to Hugging Face Hub.
+
+we can define the num_layers, num_single_layers, which determines the size of the control(default values are num_layers=4, num_single_layers=10)
+
+
+```bash
+export MODEL_DIR="black-forest-labs/FLUX.1-dev"
+export OUTPUT_DIR="path to save model"
+export TRAIN_JSON_FILE="path to your jsonl file"
+
+
+accelerate launch train_controlnet_flux.py \
+    --pretrained_model_name_or_path=$MODEL_DIR \
+    --conditioning_image_column=control_path \
+    --image_column=image_path \
+    --caption_column=caption \
+    --output_dir=$OUTPUT_DIR \
+    --jsonl_for_train=$TRAIN_JSON_FILE \
+    --mixed_precision="bf16" \
+    --resolution=512 \
+    --learning_rate=1e-5 \
+    --max_train_steps=15000 \
+    --validation_steps=100 \
+    --checkpointing_steps=200 \
+    --validation_image "./conditioning_image_1.png" "./conditioning_image_2.png" \
+    --validation_prompt "red circle with blue background" "cyan circle with brown floral background" \
+    --train_batch_size=1 \
+    --gradient_accumulation_steps=4 \
+    --report_to="tensorboard" \
+    --num_double_layers=4 \
+    --num_single_layers=0 \
+    --seed=42 \
+```
+
+To better track our training experiments, we're using the following flags in the command above:
+
+* `report_to="tensorboard` will ensure the training runs are tracked on Weights and Biases.
+* `validation_image`, `validation_prompt`, and `validation_steps` to allow the script to do a few validation inference runs. This allows us to qualitatively check if the training is progressing as expected.
+
+Our experiments were conducted on a single 40GB A100 GPU.
+
+### Inference
+
+Once training is done, we can perform inference like so:
+
+```python
+import torch
+from diffusers.utils import load_image
+from diffusers.pipelines.flux.pipeline_flux_controlnet import FluxControlNetPipeline
+from diffusers.models.controlnet_flux import FluxControlNetModel
+
+base_model = 'black-forest-labs/FLUX.1-dev'
+controlnet_model = 'path to controlnet'
+controlnet = FluxControlNetModel.from_pretrained(controlnet_model, torch_dtype=torch.bfloat16)
+pipe = FluxControlNetPipeline.from_pretrained(base_model, 
+                                              controlnet=controlnet, 
+                                              torch_dtype=torch.bfloat16)
+pipe.to("cuda")
```

Comment:
```suggestion
base_model = 'black-forest-labs/FLUX.1-dev'
controlnet_model = 'path to controlnet'
controlnet = FluxControlNetModel.from_pretrained(controlnet_model, torch_dtype=torch.bfloat16)
pipe = FluxControlNetPipeline.from_pretrained(
    base_model, 
    controlnet=controlnet, 
    torch_dtype=torch.bfloat16
)
# enable memory optimizations   
pipe.enable_model_cpu_offload()
```

Most people may not have the necessary VRAM to run it like this. So, better have it this way? WDYT?

Reply:
yes, you are right